core/client/client_builder.go

```diff
@@ -133,7 +142,12 @@ func (builder *builderStruct) Build() (*Client, error) {
 
 	client.ControlCli = builder.controllerCli
 	if client.ControlCli == nil {
-		client.ControlCli, err = newKubeClient(config, builder.useControllerDynamicMapper)
+
+		scheme := builder.scheme
+		if scheme == nil {
```

Comment:
Is scheme mandatory to build the client?

Reply:
If it is mandatory, would it be better to define it as a parameter for `NewClientBuilder()` to enforce it?

Reply:
Hey  @sutaakar! Depends on the client. I believe the CLI one would need, but overall it should not: See #742

Reply:
Ok, in such case the check should be removed.

Reply:
@sutaakar @ricardozanini Scheme is mandatory to create client. By default Controller uses `manager` object to fetch `scheme` object and initialize in client. 
https://github.com/vaibhavjainwiz/kogito-cloud-operator/blob/992fa0a55d9411fb665306682516384a8939fb40/core/client/client.go#L80

To run unit testcases, `fakeClient` uses `meta.GetRegisteredSchema()` to initialize client. 
https://github.com/vaibhavjainwiz/kogito-cloud-operator/blob/714d2b094b3e006f37b2805a5b8b2e22ad23c94a/core/test/fake_client.go#L94-L93

`Scheme` is made mandatory to create `client` as a part this PR because previously we only supposed to have one scheme for `v1beta1` which was default for client. But now we moved it from its default value to user input because product-operator will have different scheme for `v1`.

Reply:
@vaibhavjainwiz Thanks for info. I think that we could enforce the schema as parameter while creating builder to avoid possible errors like this.

Reply:
done